social/src/main/java/bisq/social/userprofile/UserProfileService.java

```diff
@@ -126,53 +135,63 @@ public void selectUserProfile(UserProfile value) {
     }*/
 
     public CompletableFuture<Optional<Entitlement.Proof>> verifyProofOfBurn(Entitlement.Type type, String proofOfBurnTxId, String pubKeyHash) {
-        // todo impl
-        //  https://github.com/bisq-network/bisq2/issues/68
-        // 
-        String userAgent = "Bisq 2";
-        //todo add providers to Bisq.config
-        List<String> providers = List.of("https://bisq.mempool.emzy.de/api/tx/",
-                "https://mempool.space/bisq/api/tx/",
-                "https://mempool.bisq.services/bisq/api/tx/");
-        String url = CollectionUtil.getRandomElement(providers);
-        Set<Transport.Type> supportedTransportTypes = networkService.getSupportedTransportTypes();
-        Transport.Type transportType;
-        if (supportedTransportTypes.contains(Transport.Type.CLEAR)) {
-            transportType = Transport.Type.CLEAR;
-        } else if (supportedTransportTypes.contains(Transport.Type.TOR)) {
-            transportType = Transport.Type.TOR;
-        } else {
-            throw new RuntimeException("I2P is not supported yet");
-        }
-        BaseHttpClient httpClient = networkService.getHttpClient(url, userAgent, transportType);
         return supplyAsync(() -> {
             try {
-                String json = httpClient.get(proofOfBurnTxId, Optional.of(new Pair<>("User-Agent", userAgent)));
-                //todo parse json 
-                boolean isBsqTx = true; //todo
-                boolean isProofOfBurn = true; //todo
-                String opReturn = pubKeyHash; //todo
-                long minBurnAmount = getMinBurnAmount(type); //todo verify if tx burned amount is >= minBurnAmount
-                if (isBsqTx && isProofOfBurn && pubKeyHash.equals(opReturn)) {
-                    return Optional.of(new Entitlement.ProofOfBurnProof(proofOfBurnTxId));
-                } else {
+                BaseHttpClient httpClient = getBsqApiHttpClient();
+                String jsonTxLookup = httpClient.get("tx/" + proofOfBurnTxId, Optional.of(new Pair<>("User-Agent", httpClient.userAgent)));
+                if (!BsqTxValidator.initialSanityChecks(proofOfBurnTxId, jsonTxLookup)) {
                     return Optional.empty();
                 }
+                boolean isBsqTx = httpClient.getBaseUrl().matches(".*//bisq.*");
+                boolean isProofOfBurn = BsqTxValidator.isProofOfBurn(jsonTxLookup);
+                byte[] publicKeyHash = Hex.decode(pubKeyHash);
+                byte[] opReturn = BsqTxValidator.getOpReturnData(jsonTxLookup);
```

Comment:
As opReturn is hex encoded we could use string comparison instead converting both to bytes